### ISSUE TITLE
[stencil] enforce linting

### ISSUE THREAD

**Issue by techsavvyash** [SamagraX-Stencil/stencil#55]

**Comment by Min2who** [comment]
do we need to add this in a github action?

**Comment by techsavvyash** [comment]
No, we need to setup linting for the stencil project that will be generated

**Comment by Min2who** [comment]
we just need to setup eslint ?

**Comment by Min2who** [comment]
> we just need to setup eslint ?

working on it

**Comment by techsavvyash** [comment]
And prettier

**Comment by Min2who** [comment]
will close it today

**Comment by techsavvyash** [comment]
Great, looking forward to the PR

**Comment by Min2who** [comment]
will close it by end of today

**Comment by Min2who** [comment]
@techsavvyash how do i run this locally as in [stencil setup docs](https://stencil.samagra.io/) it is refferencing to different repo

**Comment by techsavvyash** [comment]
> @techsavvyash how do i run this locally as in [stencil setup docs](https://stencil.samagra.io/) it is refferencing to different repo

The steps remain the same, replace the repository URL with this one. Thanks for pointing out the inconsistencies in docs, will update that

**Comment by Sandeep0306** [comment]
Hi @techsavvyash is anyone still working on this if not can you assign it to me?

**Comment by NarayanaSabari** [comment]
Hey @techsavvyash, @utnim2 was raised a pr, that was not closed till now, is there any issue with that pr, or if there is any issue with that pr, mention that, i will raise my pr

**Comment by techsavvyash** [comment]
Hey @Sandeep0306
The current PR does not quite solve for this ticket. The crux is to have a consistent and enforced linter and formatter in every project bootstrapped using stencil. So, as a part of this ticket, you'd have to raise a PR in the SamagraX-stencil/stencil-cli editing the .eslintrc and .prettierrc files and adding precommit hooks to lint and format the files on commit/save. Another additional task would be to add a GitHub actions step for linting the codebase. 
Feel free to go ahead and raise your PR, or to work collaboratively with @utnim2 in perfecting their PR. 
You can always reach out in case you face some issues.

**Comment by vsvishalsharma** [comment]
Hi @techsavvyash sir could you please confirm that do i need to update the .prettierrc file as based on this configuration, it looks like single quotes are enforced for string literals, and trailing commas are added to all elements in arrays and objects
{```

  "singleQuote": true,
  "trailingComma": "all"
}
```
also sir i am using husky for pre commit hooks am i working in the right direction

**Comment by techsavvyash** [comment]
Hey @vsvishalsharma,
Yes you are in the right direction. Thanks for the PR, I have left a comment on your PR, it would be great if you could take that into account and raise the required PR as well, will merge both of them together and assign this one to you.
